lib/conjur/graph.rb

```diff
@@ -144,7 +144,11 @@ def vertices
 
     def node_id_for role
       role = role.id if role.respond_to?(:id)
-      @node_ids[role]
+      node_id = @node_ids[role]
+      if node_id.nil?
+        node_id = @node_ids[role] = next_node_id
+      end
+      node_id
```

Comment:
I don't understand how is this code more efficient. It appears equivalent (except for the handling of nil values), only more convoluted, and, if anything, slower, because it causes an additional allocation (regardless of the node existence).